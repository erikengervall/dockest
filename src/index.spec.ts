import * as dockestExport from './index'

describe('Dockest', () => {
  it('should export the expected members', () => {
    expect(dockestExport).toEqual({
      default: expect.any(Function),
      execa: expect.any(Function),
      logLevel: {
        DEBUG: expect.any(Number),
        ERROR: expect.any(Number),
        INFO: expect.any(Number),
        NOTHING: expect.any(Number),
        WARN: expect.any(Number),
      },
      runners: {
        KafkaRunner: expect.any(Function),
        PostgresRunner: expect.any(Function),
        RedisRunner: expect.any(Function),
        ZooKeeperRunner: expect.any(Function),
        SimpleRunner: expect.any(Function),
      },
      sleep: expect.any(Function),
    })
  })

  it('should be initializable and expose the main run method', () => {
    const Dockest = dockestExport.default
    const RedisRunner = dockestExport.runners.RedisRunner
    const redisRunner = new RedisRunner({ service: '_', image: '_' })

    const dockest = new Dockest({ runners: [redisRunner] })

    expect(dockest).toBeInstanceOf(Dockest)
    expect(dockest).toEqual(
      expect.objectContaining({
        config: expect.objectContaining({
          runners: expect.arrayContaining([redisRunner]),
        }),
        run: expect.any(Function),
      }),
    )
  })
})
