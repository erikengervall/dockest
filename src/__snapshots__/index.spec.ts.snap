// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dockest should be initializable and expose the main run method 1`] = `
Dockest {
  "attachRunners": [Function],
  "config": Object {
    "$": Object {
      "failedTeardowns": Array [],
      "jestRanWithResult": false,
      "perfStart": 0,
    },
    "jest": Object {},
    "opts": Object {
      "afterSetupSleep": 0,
      "composeFile": Array [],
      "dev": Object {
        "debug": false,
      },
      "dumpErrors": false,
      "exitHandler": null,
      "logLevel": 3,
      "runInBand": true,
    },
    "runners": Array [
      GeneralPurposeRunner {
        "containerId": "",
        "getComposeService": [Function],
        "initializer": "",
        "logger": Logger {
          "debug": [Function],
          "error": [Function],
          "info": [Function],
          "runnerService": "_",
          "runnerSymbol": "ðŸ¦‡ ",
          "setRunnerSymbol": [Function],
          "warn": [Function],
        },
        "runnerConfig": Object {
          "build": "",
          "commands": Array [],
          "connectionTimeout": 5,
          "dependsOn": Array [],
          "host": "localhost",
          "image": "_",
          "networks": undefined,
          "ports": Object {},
          "props": Object {},
          "responsivenessTimeout": 30,
          "service": "_",
        },
      },
    ],
  },
  "run": [Function],
  "validateConfig": [Function],
}
`;

exports[`Dockest should export the expected members 1`] = `
Object {
  "default": [Function],
  "execa": [Function],
  "logLevel": Object {
    "DEBUG": 4,
    "ERROR": 1,
    "INFO": 3,
    "NOTHING": 0,
    "WARN": 2,
  },
  "runners": Object {
    "GeneralPurposeRunner": [Function],
    "KafkaRunner": [Function],
    "PostgresRunner": [Function],
    "RedisRunner": [Function],
    "ZooKeeperRunner": [Function],
  },
  "sleep": [Function],
}
`;
